frontend/src/v5/ui/components/viewer/drawingViewer/viewerLayer2D/viewerLayer2D.component.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { CSSProperties, useEffect, useRef, useState } from 'react';
+import { Container, LayerLevel } from './viewerLayer2D.styles';
+import { PanZoomHandler } from '../panzoom/centredPanZoom';
+import { isEqual } from 'lodash';
+import { Offset, SvgArrow } from './svgArrow/svgArrow.component';
+import { SvgCircle } from './svgCircle/svgCircle.component';
+
+export type ViewBoxType = ReturnType<PanZoomHandler['getOriginalSize']> & ReturnType<PanZoomHandler['getTransform']>;
+type ViewerLayer2DProps = {
+	viewBox: ViewBoxType,
+	active: boolean,
+	onChange?: (arrow: { start: Offset, end: Offset }) => void;
+};
+export const ViewerLayer2D = ({ viewBox, active, onChange }: ViewerLayer2DProps) => {
+	const [offsetStart, setOffsetStart] = useState<Offset>(null);
+	const [offsetEnd, setOffsetEnd] = useState<Offset>(null);
+	const previousViewBox = useRef<ViewBoxType>(null);
+	const [mousePosition, setMousePosition] = useState<Offset>(null);
+
+	const layerStyle: CSSProperties = {
+		transformOrigin: '0 0',
+		transform: `matrix(${viewBox.scale}, 0, 0, ${viewBox.scale}, ${viewBox.x}, ${viewBox.y})`,
+		width: `${viewBox.width}px`,
+		height: `${viewBox.height}px`,
+	};
+
+	// This returns the offset of the cursor from the top-left corner
+	const getCursorOffset = (e) => {
+		const rect = e.target.getBoundingClientRect();
+		const offsetX = e.clientX - rect.left;
+		const offsetY = e.clientY - rect.top;
+		return [offsetX, offsetY].map((point) => point / viewBox.scale) as Offset;
+	};
+
+	const handleMouseDown = () => previousViewBox.current = viewBox;
```

Comment:
It takes a couple of clicks before starting to draw
![drawing arrow](https://github.com/3drepo/3drepo.io/assets/42032433/c5f24a5e-ecc8-4024-91a6-9f8566206788)